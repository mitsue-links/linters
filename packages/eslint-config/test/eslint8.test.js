/* eslint strict: [2, "global"] */
'use strict';

const path = require('path');
const {lintFile} = require('./helper');

describe('ESLint 8.0.0', () => {
    test('8.0.0_ok.js', async () => {
        const filePath = '8.0.0_ok.js';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('8.0.0_error.js', async () => {
        const filePath = '8.0.0_error.js';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});
