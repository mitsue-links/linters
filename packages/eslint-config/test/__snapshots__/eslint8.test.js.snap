// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint 8.0.0 8.0.0_error.js 1`] = `
[
  {
    "filePath": "8.0.0_error.js",
    "messages": [
      {
        "line": 3,
        "ruleId": "class-methods-use-this",
        "severity": 1,
      },
      {
        "line": 7,
        "ruleId": "class-methods-use-this",
        "severity": 1,
      },
    ],
  },
]
`;

exports[`ESLint 8.1.0 8.1.0_error.js 1`] = `
Array [
  Object {
    "filePath": "8.1.0_error.js",
    "messages": Array [
      Object {
        "line": 6,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
      Object {
        "line": 8,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
      Object {
        "line": 12,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
      Object {
        "line": 14,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`ESLint 8.5.0 8.5.0_error.js 1`] = `
Array [
  Object {
    "filePath": "8.5.0_error.js",
    "messages": Array [
      Object {
        "line": 12,
        "ruleId": "prefer-object-has-own",
        "severity": 2,
      },
    ],
  },
]
`;
